src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -0,0 +1,321 @@
+import React, { useContext } from 'react'
+import { add, addDays, addMonths, format, isToday } from 'date-fns'
+import { CardTitle, LayoutContext, LoadingIndicator } from '..'
+import { Area, AreaChart, Bar, BarChart, CartesianGrid, Cell, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import "./TimeSeriesChart.css"
+import { AxisDomain } from 'recharts/types/util/types'
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import getDaysInMonth from 'date-fns/getDaysInMonth'
+import { ceil } from 'lodash'
+import addHours from 'date-fns/addHours'
+
+export interface TimeSeriesChartProps {
+    title?: string
+    intervalType?: "Week" | "Month" | "SixMonth" | "Day",
+    intervalStart: Date,
+    data: any[] | undefined,
+    dataGap?: Duration,
+    dataKeys?: string[],
+    chartHasData: boolean,
+    tooltip: ({ active, payload, label }: any) => React.JSX.Element | null,
+    chartType: "Line" | "Bar" | "Area"
+    options?: LineChartOptions | BarChartOptions | AreaChartOptions
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export interface LineChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    connectNulls?: boolean,
+    domainMin?: number | "Auto"
+}
+
+export interface BarChartOptions {
+    barColor?: ColorDefinition | ColorDefinition[],
+    thresholds?: BarChartThreshold[]
+}
+
+export interface BarChartThreshold {
+    value: number
+    referenceLineColor?: ColorDefinition
+    overThresholdBarColor?: ColorDefinition
+}
+
+export interface AreaChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    areaColor?: ColorDefinition | ColorDefinition[],
+}
+
+export default function TimeSeriesChart(props: TimeSeriesChartProps) {
```

Comment:
First glance looking at the storybook, this looks quite nice and robust!  I had to fiddle with the axes a lot to get the bar/area charts working right, but looking at the stories it seems largely consistent.